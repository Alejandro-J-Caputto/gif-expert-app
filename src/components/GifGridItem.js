import React from 'react'
import PropTypes from 'prop-types';


export const GifGridItem = ({ id, title, url }) => {

    return (
        <div className="card animate__animated animate__fadeIn">
            <img src={url} alt={title} />
            <p key={id}>{title}</p>
        </div>
    )
}



GifGridItem.protoTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired
}

/*

Enzyme
Enzyme to JSON

 */

