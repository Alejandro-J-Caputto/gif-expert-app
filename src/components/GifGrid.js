
import React from 'react'
import PropTypes from 'prop-types';
import { GifGridItem } from './GifGridItem'
// import { getGifs } from '../helpers/getGifs'
import { useFetchGifs } from '../hooks/useFetchGifs'


export const GifGrid = ({ categorie }) => {



    // const [images, setImages] = useState([])

    // useEffect(() => {
    //     getGifs(categorie).then(gifs => {
    //         setImages(gifs)
    //     })
    // }, [category])


    //APLICANDO HOOK

    const { data: images, loading } = useFetchGifs(categorie)

    /////////////
    return (
        <>
            <h3 className="animate__animated animate__bounce">{categorie}</h3>
            {loading && <p>Loading</p>}

            {
                <div className="card-grid animate__animated animate__fadeIn">

                    {images.map(gif => {
                        return <GifGridItem
                            key={gif.id}
                            {...gif}

                        />

                    })}


                </div>
            }
        </>
    )
}
GifGrid.propTypes = {
    categorie: PropTypes.string.isRequired
}