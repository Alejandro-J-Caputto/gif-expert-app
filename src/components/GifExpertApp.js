

import React, { useState } from 'react';
import PropTypes from 'prop-types';
import { AddCategory } from './AddCategory';
import { GifGrid } from './GifGrid';


export const GifExpertApp = () => {

    const [categorias, setcategorias] = useState(['One Punch'])
    // const handleAdd = () => {
    //     // setcategorias([...categorias, 'Naruto'])
    //     setcategorias(cat =>
    //         [...categorias, 'Bleach']
    //     )
    // };


    return (
        <>
            <h1>GifExpertApp</h1>
            <AddCategory setcategorias={setcategorias} />
            <hr />

            {/* <button onClick={handleAdd} > agregar </button> */}
            <ol>
                {
                    categorias.map((categorie) => {
                        return <GifGrid key={categorie} categorie={categorie} />
                    })
                }
            </ol>
        </>
    )
}

export default GifExpertApp;