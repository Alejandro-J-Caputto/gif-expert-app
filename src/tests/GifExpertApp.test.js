import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import { GifExpertApp } from '../components/GifExpertApp'

describe('Pruebas en componente principal gifExpertApp', () => {
    const wrapper = shallow(<GifExpertApp />)
    test('Probar snapShot de GifexpertApp', () => {

        expect(wrapper).toMatchSnapshot();
    });

    test('debe de mostrar una lista de categorias', () => {
        const categories = ['One Punch', 'Dragon Ball'];
        const wrapper = shallow(<GifExpertApp defaultCategories={categories} />);

        expect(wrapper).toMatchSnapshot()
        expect(wrapper.find('GifGrid')).toMatchSnapshot(categories.length);

    })


})
