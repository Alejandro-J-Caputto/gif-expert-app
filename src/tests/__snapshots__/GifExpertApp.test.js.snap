// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en componente principal gifExpertApp Probar snapShot de GifexpertApp 1`] = `
<Fragment>
  <h1>
    GifExpertApp
  </h1>
  <AddCategory
    setcategorias={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;

exports[`Pruebas en componente principal gifExpertApp debe de mostrar una lista de categorias 1`] = `
<Fragment>
  <h1>
    GifExpertApp
  </h1>
  <AddCategory
    setcategorias={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      categorie="One Punch"
      key="One Punch"
    />
    <GifGrid
      categorie="Dragon Ball"
      key="Dragon Ball"
    />
  </ol>
</Fragment>
`;

exports[`Pruebas en componente principal gifExpertApp debe de mostrar una lista de categorias 2`] = `
Array [
  <GifGrid
    categorie="One Punch"
    key="One Punch"
  />,
  <GifGrid
    categorie="Dragon Ball"
    key="Dragon Ball"
  />,
]
`;
